cmd/juju/model/configcommand.go

```diff
@@ -63,6 +65,70 @@ func NewConfigCommand() cmd.Command {
 
 type attributes map[string]interface{}
 
+// CoerceFormat attempts to convert the attributes values from the complex type
+// to the more simple type. This is because the output of this command outputs
+// in the following format:
+//
+//     resource-name:
+//        value: foo
+//        source: default
+//
+// Where the consuming side of the command expects it in the following format:
+//
+//     resource-name: foo
+//
+// CoerceFormat attempts to diagnose this and attempt to do this correctly.
+func (a attributes) CoerceFormat() (attributes, error) {
+	coerced := make(map[string]interface{})
+
+	fields := schema.FieldMap(schema.Fields{
+		"value":  schema.Any(),
+		"source": schema.String(),
+	}, nil)
+
+	for k, v := range a {
+		out, err := fields.Coerce(v, []string{})
+		if err != nil {
+			// Fallback to the old format and just pass through the value.
+			coerced[k] = v
+			continue
+		}
+
+		m := out.(map[string]interface{})
+		v = m["value"]
+
+		// Resource tags in the new output format is a map[string]interface{},
+		// but it should be of the format `foo=bar baz=boo`.
+		if k == "resource-tags" {
+			tags, err := coerceResourceTags(v)
+			if err != nil {
+				return nil, errors.Annotate(err, "unable to read resource-tags")
+			}
+			v = tags
+		}
+
+		coerced[k] = v
```

Comment:
Is it worth detecting `agent-version` and dropping it with a warning? This way you don't have to resort to sed magic when doing and export -> import pass.

Reply:
Maybe if we had `--ignore-agent-version` as a flag or similar.